quarkchain/cluster/jsonrpc.py

```diff
@@ -1277,6 +1274,30 @@ async def setTargetBlockTime(self, root_block_time=0, minor_block_time=0):
             root_block_time, minor_block_time
         )
 
+    @public_methods.add
+    @decode_arg("block_hash", hash_decoder)
+    @decode_arg("token_id", quantity_decoder)
+    @decode_arg("address", address_decoder)
+    async def getTotalBalance(self, block_hash, token_id, address, limit=100):
+        try:
+            starter = Address.create_from(address)
+            full_shard_id = self.master.env.quark_chain_config.get_full_shard_id_by_full_shard_key(
+                starter.full_shard_key
+            )
+            result = await self.master.get_total_balance(
+                Branch(full_shard_id), block_hash, token_id, starter.recipient, limit,
+            )
+            total_balance, next_starter = result
+            return {
+                "totalBalance": quantity_encoder(total_balance),
+                "next_addr": data_encoder(
+                    next_starter + (starter.full_shard_key).to_bytes(4, byteorder="big")
+                ),
+            }
+        except Exception as e:
+            Logger.log_exception()
+            raise e
```

Comment:
how about raising invalid request?

Reply:
actually, best practice is, raising invalid request if branch not found, while raising serve error if network fails (during the async/await call in master)